### DISCUSSION TITLE
Number string lengths

### DISCUSSION THREAD

**Discussion by HappyPathAB** [Labs64/NetLicensing-Community#73] [Questions] [answered]
Various entities have identities referred to as "numbers" in the documentation, although they're actually strings.

Is there a limitation on the (significant) length of these strings, and if so what?

In other words, is there a point at which the API will refuse to accept a request because a string is too long, **or** is there a point at which only the first X characters of two "number" strings will be used to compare them?

I'm not after a theoretical maximum here but a practical one. If it's less than 256, say, that's important to know.

**Comment by r-brown** [comment, marked as answer]
That's correct - all entities in the NetLicensing can be refrred using a functional unique number.
These numbers can be defined:
- as random number generated by NetLicensing (default)
  - current format is defined as follows: `<PREFIX><8-CHAR>`; e.g. `P12345678`
- set by vendor on entity create or update methods
  - Some practical examples here:
    - Customer Number from the CRM system
    - Product number from the ERP
    - SKU number from the eCommerce system

---

The max length for all numbers is set 1000ch.

**Comment by r-brown** [comment]
### Did you know!

- the nature of all entities numbers is functional, therefore these numbers can be also made available/shared with the end-customers; e.g. `LicenseeNumber` (customer number)
- we do not include following charactes in the numbers generated by NetLicensing: `0O1Ll8B` ; these characters are proven to be missinterpreted while typing - will save uncessary back-and-forth on the customer support side ;-)
